tests/Umbraco.Tests.Integration/Umbraco.Examine.Lucene/UmbracoExamine/BackOfficeExamineSearcherTests.cs

```diff
@@ -0,0 +1,647 @@
+        Assert.AreEqual(actual.First().Values["nodeName"], contentName);
+        Assert.AreEqual(actual.First().Id, contentId);
+    }
+
+    [Test]
+    public async Task Search_Two_Published_Content_With_Similar_Names_By_Name()
+    {
+        // Arrange
+        SetupUserIdentity(Constants.Security.SuperUserIdAsString);
+
+        string contentName = "TestName Original";
+        string secondContentName = "TestName Copy";
+
+        var contentType = new ContentTypeBuilder()
+            .WithId(0)
+            .Build();
+        ContentTypeService.Save(contentType);
+
+        var firstContent = new ContentBuilder()
+            .WithId(0)
+            .WithName(contentName)
+            .WithContentType(contentType)
+            .Build();
+        await ExecuteAndWaitForIndexing(() => ContentService.SaveAndPublish(firstContent), Constants.UmbracoIndexes.InternalIndexName);
+
+        var secondContent = new ContentBuilder()
+            .WithId(0)
+            .WithName(secondContentName)
+            .WithContentType(contentType)
+            .Build();
+        await ExecuteAndWaitForIndexing(() => ContentService.SaveAndPublish(secondContent), Constants.UmbracoIndexes.InternalIndexName);
+
+        // IndexRebuilder.RebuildIndex("InternalIndex");
+        string query = contentName;
+
+        // Act
+        IEnumerable<ISearchResult> actual = BackOfficeExamineSearch(query);
+
+        // Assert
+        Assert.AreEqual(2, actual.Count());
+        // Checks if the first content in the search is the original content
+        Assert.AreEqual(actual.First().Id, firstContent.Id.ToString());
+        // Checks if the score for the original name is higher than the score for the copy
+        Assert.Greater(actual.First().Score, actual.Last().Score);
+
+        Thread.Sleep(3000);
+    }
+
+    [Test]
+    public async Task Search_For_Child_Published_Content_With_Query_By_Content_Name()
+    {
+        // Arrange
+        SetupUserIdentity(Constants.Security.SuperUserIdAsString);
+
+        string contentName = "ParentTestContent";
+        string childContentName = "ChildTestContent";
+
+        var contentType = new ContentTypeBuilder()
+            .WithName("Document")
+            .Build();
+        ContentTypeService.Save(contentType);
+
+        var content = new ContentBuilder()
+            .WithName(contentName)
+            .WithContentType(contentType)
+            .Build();
+        await ExecuteAndWaitForIndexing(() => ContentService.SaveAndPublish(content), Constants.UmbracoIndexes.InternalIndexName);
+
+        var childContent = new ContentBuilder()
+            .WithName(childContentName)
+            .WithContentType(contentType)
+            .WithParentId(content.Id)
+            .Build();
+        await ExecuteAndWaitForIndexing(() => ContentService.SaveAndPublish(childContent), Constants.UmbracoIndexes.InternalIndexName);
+
+
+        IndexRebuilder.RebuildIndex("InternalIndex");
```

Comment:
You seem to be doing this in quite a few tests, but I'm not sure why it's necessary? SaveAndPublish should index the values, so it should be nececary to rebuild the indexes as well.